packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -134,54 +123,47 @@ export default createRule<Options, MessageIds>({
         caseTypes.add(caseType);
       }
 
-      const unionTypes = tsutils.unionTypeParts(discriminantType);
-      const missingBranchTypes = unionTypes.filter(
-        unionType => !caseTypes.has(unionType),
-      );
+      const missingLiteralBranchTypes: ts.Type[] = [];
+
+      for (const unionPart of tsutils.unionTypeParts(discriminantType)) {
+        for (const intersectionPart of tsutils.intersectionTypeParts(
+          unionPart,
+        )) {
```

Comment:
[Non-Actionable] This feels like something we'd want a shared utility for... and, ha, I filed an issue for this back in August! https://github.com/typescript-eslint/typescript-eslint/pull/7466#discussion_r1295028416 -> https://github.com/JoshuaKGoldberg/ts-api-utils/issues/258

Not requesting changes here. I'll go ahead and send that PR to `ts-api-utils` and then follow up in this repo.

Reply:
https://github.com/JoshuaKGoldberg/ts-api-utils/pull/374